### PR TITLE
PublicDashboards: Configuration modal redesign

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

Redesign for configuration modal

**Which issue(s) does this PR fix?**:

Fixes [#4682](https://github.com/grafana/grafana-enterprise/issues/4682)

**Special notes for your reviewer**:

**Before**

https://user-images.githubusercontent.com/11707172/217885285-3d370325-1347-491f-9bb4-cdfc2f4eff9a.mov

**After**

https://user-images.githubusercontent.com/11707172/217887902-25c2474c-518c-4004-9568-44857eb95321.mov

### CODE REVIEW

**Comment by polibb** [comment]
When I change the time range of the whole dashboard to something other than 6 hours, and go into sharing publicly, the sharing option is still on "Last 6 hours". 

image

File: e2e/dashboards-suite/dashboard-public-create.spec.ts [hunk 1]
```diff
@@ -106,29 +121,20 @@ e2e.scenario({
...
     e2e().intercept('PUT', '/api/dashboards/uid/ZqZnVvFZz/public-dashboards/*').as('update');
-    e2e.pages.ShareDashboardModal.PublicDashboard.SaveConfigButton().click();
+    // Switch off enabling toggle
+    e2e.pages.ShareDashboardModal.PublicDashboard.PauseSwitch().should('be.enabled').click({ force: true });
     e2e().wait('@update');
 
     // Url should be hidden
-    e2e.pages.ShareDashboardModal.PublicDashboard.CopyUrlInput().should('not.exist');
+    e2e.pages.ShareDashboardModal.PublicDashboard.CopyUrlInput().should('be.disabled');
+    e2e.pages.ShareDashboardModal.PublicDashboard.CopyUrlButton().should('be.disabled');
```
Comment (polibb) [inline e2e/dashboards-suite/dashboard-public-create.spec.ts:137]: Great test coverage! 👏🏻

**Comment by juanicabanas** [comment]
> When I change the time range of the whole dashboard to something other than 6 hours, and go into sharing publicly, the sharing option is still on "Last 6 hours".
> 
> image

that's because you didn't save the current time range as dashboard default. Once you do it, the change will be shown

**Review by kaydelaney — APPROVED** [review]
Looks good!

Extremely small nitpick, but I think some slight adjustments could be made to make everything perfectly center-aligned:

Before:
image


After (Notice the info icon):
image

If you'd like I could push the changes required for this (or if you think it looks better as it is now, that's totally fine too!)

**Comment by juanicabanas** [comment]
> d for this (or if you think it looks better as it is

sure! make the change yourself 💪🏽 thanks for the detail, looks quite better

**Review by polibb — APPROVED** [review]

**Comment by kaydelaney** [comment]
@juanicabanas Yep! Will do it now

**Comment by juanicabanas** [comment]
@grafana/dataviz-squad could you take a look at Checkbox modifications? thanks!

**Comment by staton-hyse11** [comment]
LGTM!

**Pull request merged**